src/sandstorm/run-bundle.c++

```diff
@@ -921,8 +921,15 @@ private:
     return kj::heapString(exeName.slice(0, KJ_ASSERT_NONNULL(exeName.findLast('/'))));
   }
 
-  void changeToInstallDir() {
+  void changeToInstallDirAndCheckAccess() {
     KJ_SYSCALL(chdir(getInstallDir().cStr()));
+    if (access("../var/sandstorm", W_OK) == -1) {
+      if (errno == EACCES) {
+        KJ_FAIL_REQUIRE("Sandstorm must be run as root or a member of SERVER_USER's group.");
```

Comment:
However, `dev` should never run as root, even if Sandstorm was installed as root; it should be run by someone inside the server user's group.